unit_tests/test_redis_client_base.py

```diff
@@ -0,0 +1,237 @@
+
+    @staticmethod
+    @patch("d3a_api_client.redis_client_base.wait_until_timeout_blocking",
+           side_effect=AssertionError)
+    def test_unregister_is_active_true_throws_exception(
+            mock_wait_until_timeout_blocking,
+            redis_client_auto_register):
+        """Check if is active set to true throws exception."""
+        with pytest.raises(RedisAPIException,
+                           match="API unregister process timed out. "
+                                 "Server will continue processing "
+                                 "your request on the background and "
+                                 "will notify you as soon as "
+                                 "the unregistration has been completed."):
+            redis_client_auto_register.is_active = True
+            redis_client_auto_register.unregister(is_blocking=True)
+        mock_wait_until_timeout_blocking.assert_called()
+
+    @staticmethod
+    def test_on_register(redis_client_auto_register):
+        """Check the on_register function with correct message that doesn't throw exception."""
+        data = {"device_uuid": DEVICE_ID, "transaction_id": TRANSACTION_ID}
+        message = {"data": json.dumps(data)}
+        redis_client_auto_register._blocking_command_responses = {
+            "register": {"transaction_id": TRANSACTION_ID}}
+        redis_client_auto_register._on_register(message)
+        assert redis_client_auto_register.is_active is True
+        assert redis_client_auto_register.area_uuid == DEVICE_ID
+
+    @staticmethod
+    def test_on_unregister(redis_client_auto_register):
+        """Check the on_unregister function with correct message that doesn't throws exception."""
+        data = {"device_uuid": DEVICE_ID, "transaction_id": TRANSACTION_ID,
+                "response": "success"}
+        message = {"data": json.dumps(data)}
+        redis_client_auto_register._blocking_command_responses = {
+            "unregister": {"transaction_id": TRANSACTION_ID}}
+        redis_client_auto_register._on_unregister(message)
+        assert redis_client_auto_register.is_active is False
+
+    @staticmethod
+    def test_on_unregister_throws_exception(redis_client_auto_register):
+        """Check if exception is raised when response of on_unregister is not successful."""
+        with pytest.raises(RedisAPIException,
+                           match=f"Failed to unregister from market {AREA_ID}."
+                                 "Deactivating connection."):
+            data = {"device_uuid": DEVICE_ID, "transaction_id": TRANSACTION_ID,
+                    "response": "unsuccessful"}
+            message = {"data": json.dumps(data)}
+            redis_client_auto_register._blocking_command_responses = {
+                "unregister": {"transaction_id": TRANSACTION_ID}}
+            redis_client_auto_register._on_unregister(message)
+            assert redis_client_auto_register.is_active is True
```

Comment:
This line is not doing anything. As we've discussed previously, the test will exit its context (`with pytest.raises(...)` as soon as an exception is raised. Therefore, the execution will never reach your final assertion.
Moreover: if you fix it (as we've done with the other tests) you'll actually see that this assertion is wrong and it fails.

Reply:
Yes, you are correct. I will remove the last line.